src/ai/ai_gui.cpp

```diff
@@ -1176,7 +1176,7 @@ struct AIDebugWindow : public Window {
 			this->autoscroll = this->vscroll->GetPosition() >= log->used - this->vscroll->GetCapacity();
 		}
 		if (this->autoscroll) {
-			int scroll_pos = max(0, log->used - this->vscroll->GetCapacity());
+			uint scroll_pos = (this->vscroll->GetCapacity() > log->used ? 0 : log->used - this->vscroll->GetCapacity());
```

Comment:
In general, and that goes for all the next instances here too, I have a hard time processing the direction of these statements. I would expect:

`uint scroll_pos = (log->used < this->vscroll->GetCapacity() ? 0 : log->used - this->vscroll->GetCapacity());`

As than both statements read in the same order. This flipping around requires more processing power in my brain. But I fully admit that this can be a local problem (as in: in my brain problem), but maybe something to consider :)

(and yes, the inverse of your statement is `<=`, but in this case that totally doesn't matter .. I guess for both our cases, adding the `=` would be a slight reduction of CPU time :P)

Reply:
No, I take this back; I write both forms when I am writing code myself. Seems the order just depends on how/when I am reading it. This is really weird, how brains work :D